### DISCUSSION TITLE
Display env variable in help text for flags

### DISCUSSION THREAD

**Discussion by gandro** [pacak/bpaf#198] [Q&A] [answered]
Thanks a lot for this wonderfully crafted crate, it's a pleasure to use.

I noticed that while `switch()` and `flag()` seem to support env variable fallback, the `env` variable is not shown in the help output as it is for arguments.

For a bit more context: In my application, I have two options. One allows the user to specify a font directory via path, the other one just tells the application to search for fonts in system directories. I want both options to be have an environment variable fallback, and this seems to work correctly (i.e. the `env()` call is functional). *However*, in the generated help output, the environment variable for the flag is not displayed, making it hard for users to discover the env fallback for the `system-fonts` option.

Here's are the two bpaf parsers:

```rust
let fonts_dir = long("fonts")
    .env("OIKOS_FONTS")
    .help("Load fonts from this directory")
    .argument::<PathBuf>("DIR")
    .optional();
let system_fonts = long("system-fonts")
    .env("OIKOS_SYSTEM_FONTS")
    .help("Search for additional fonts in system directories")
    .switch();
```

And here's the currently generated help output:
```
        --fonts <DIR>    [env:OIKOS_FONTS: N/A]
                         Load fonts from this directory
        --system-fonts   Search for additional fonts in system directories
```

Where I would love to be able to see something like this:
```
        --fonts <DIR>    [env:OIKOS_FONTS: N/A]
                         Load fonts from this directory
        --system-fonts   [env:OIKOS_SYSTEM_FONTS: N/A]
                         Search for additional fonts in system directories
```

Would something like this be possible?

**Comment by pacak** [comment, marked as answer]
It's a bug, fixed in https://github.com/pacak/bpaf/pull/200

I changed the representation from your desired output a bit though - it doesn't really matter what the value of the env variable is for flag purposes. bpaf checks if it's set or unset so it would look like this:

```text
Usage: [--fonts DIR] [--system-fonts]
Available options:
        --fonts <DIR>   [env:OIKOS_FONTS: N/A]
                        Load fonts from this directory
        --system-fonts  [env:OIKOS_SYSTEM_FONTS: not set]
                        Search for additional fonts in system directories
    -h, --help          Prints help information
```

Thank you for the report, I'll publish this change along with a few other open tickets I have a bit later.

**Reply by gandro** [reply]
Awesome, thank you!

I fully agree with the "not set" displaying part being more aligned with how it works.
